### PR TITLE
GMA  AdView::ad_size()

### PR DESCRIPTION
**Pull request by DellaBitta**
Add a method to retrieve the AdSize that was used to initialize the AdView.

***

[Integration Test CI](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1786111124)
[Packaging CI](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1786111919) -> [Integration Test CI](https://github.com/firebase/firebase-cpp-sdk/actions/runs/1786633499)

***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [X] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: gma/src/android/ad_view_internal_android.h [hunk 1]
```diff
@@ -84,27 +84,31 @@ class AdViewInternalAndroid : public AdViewInternal {
...
+  // Convenience method to "dry" the JNI calls that don't take parameters beyond
+  // the future callback pointer.
+  Future<void> InvokeNullary(AdViewFn fn, ad_view_helper::Method method);
 
   // Reference to the Android AdView object used to display AdView ads.
   jobject ad_view_;
 
+  // The AdSize that was used to initialize the ad.
+  AdSize ad_size_;
+
```
Comment (DellaBitta) [inline gma/src/android/ad_view_internal_android.h:99]: This is the only actual change here ^

File: gma/src/android/ad_view_internal_android.h [hunk 2]
```diff
@@ -84,27 +84,31 @@ class AdViewInternalAndroid : public AdViewInternal {
   Future<void> Resume() override;
   Future<void> Destroy() override;
   bool is_initialized() const override { return initialized_; }
+  AdSize ad_size() const override { return ad_size_; }
 
  private:
-  // Reference to the Java helper object used to interact with the Mobile Ads
```
Comment (DellaBitta) [inline gma/src/android/ad_view_internal_android.h:89]: Reordered the members.  The actual change is denoted below.

File: gma/src/ios/ad_view_internal_ios.h [hunk 3]
```diff
@@ -62,23 +63,26 @@ class AdViewInternalIOS : public AdViewInternal {
   /// Contains information to asynchronously complete the LoadAd Future.
   FutureCallbackData<AdResult>* ad_load_callback_data_;
 
-  /// Prevents duplicate invocations of initailize on the AdView.
-  bool initialized_;
+  // The AdSize that was used to initialize the ad.
+  AdSize ad_size_;
```
Comment (DellaBitta) [inline gma/src/ios/ad_view_internal_ios.h:67]: This is the real change here. Other members were reordered.

File: gma/integration_test/src/integration_test.cc [hunk 4]
```diff
@@ -987,6 +987,44 @@ TEST_F(FirebaseGmaTest, TestAdViewLoadAdDestroyNotCalled) {
   WaitForCompletion(ad_view->LoadAd(GetAdRequest()), "LoadAd");
   delete ad_view;
 }
+
+TEST_F(FirebaseGmaTest, TestAdViewAdSizeCompareOp) {
+  using firebase::gma::AdSize;
+  EXPECT_TRUE(AdSize(50, 100) == AdSize(50, 100));
+  EXPECT_TRUE(AdSize(100, 50) == AdSize(100, 50));
```
Comment (jonsimantov) [inline gma/integration_test/src/integration_test.cc:994]: maybe throw in for good measure...
EXPECT_FALSE(AdSize(50, 100) == AdSize(100, 50))

**Review by jonsimantov — APPROVED** [review]
Is the ad_size_ actually used anywhere? Or did that code already exist?

File: gma/integration_test/src/integration_test.cc (hunk 4 above)
Comment (DellaBitta) [reply gma/integration_test/src/integration_test.cc:994]: Done!  Added some != tests, too.

**Pull request merged**